ansible_collections/arista/avd/roles/eos_validate_state/templates/validate_state_report-csv.j2

```diff
@@ -1,230 +1,28 @@
-{%     endif %}
-{% endfor %}
-{# Routing table check: loopback0 Results #}
-{% for node in groups[fabric_name] | arista.avd.natural_sort %}
-{%     if hostvars[node].routing_table_loopback0_results is defined and hostvars[node].routing_table_loopback0_results.results is defined %}
-{%         for result in hostvars[node].routing_table_loopback0_results.results %}
-{%             if result.skipped is not defined %}
-{%                 set test_id.value = test_id.value + 1 %}
-{{ test_id.value }},{{ node }},Routing Table,Remote Lo0 address,{{ result.routing_table_loopback0_test.loopback0_address }},{% if result.failed == false %}PASS{% else %}FAIL,{{ result.msg }}{% endif %}
-
-{%             endif %}
-{%         endfor %}
-{%     endif %}
-{% endfor %}
-{# loopback0 reachability Results #}
-{% for node in groups[fabric_name] | arista.avd.natural_sort %}
-{%     if hostvars[node].loopback0_reachability_results is defined and hostvars[node].loopback0_reachability_results.results is defined %}
-{%         for result in hostvars[node].loopback0_reachability_results.results %}
-{%             if result.skipped is not defined %}
-{%                 set test_id.value = test_id.value + 1 %}
-{{ test_id.value }},{{ node }},Loopback0 Reachability,Loopback0 Reachability,Source: {{ node }} - {{ hostvars[node].loopback_interfaces.Loopback0.ip_address | ansible.netcommon.ipaddr('address') }} Destination: {{ result.loopback0_reachability_test.loopback0_address }},{% if result.failed == false %}PASS{% else %}FAIL,{{ result.msg }}{% endif %}
-
-{%             endif %}
-{%         endfor %}
-{%     endif %}
-{% endfor %}
-{# Inb management to loopback 0 reachability Results #}
-{% for node in groups[fabric_name] | arista.avd.natural_sort %}
-{%     if hostvars[node].inb_mgmt_loopback0_reachability_results.results is arista.avd.defined %}
-{%         for result in hostvars[node].inb_mgmt_loopback0_reachability_results.results %}
-{%             if result.skipped is not defined %}
-{%                 set test_id.value = test_id.value + 1 %}
-{{ test_id.value }},{{ node }},Loopback0 Reachability,Inband Mgmt Reachability,Source: {{ node }} - {{ result.inb_mgmt_loopback0_reachability_test.invocation.module_args.commands[0].split(" ") | last }} Destination: {{ result.inb_mgmt_loopback0_reachability_test.loopback0_address }},{% if result.failed == false %}PASS{% else %}FAIL,{{ result.msg }}{% endif %}
-
-{%             endif %}
-{%         endfor %}
+{% for report_entry in eos_validate_state_report | arista.avd.default([]) %}
+{%     if only_failed_tests is arista.avd.defined(false) %}
+{%         set csv_line = [] %}
+{%         do csv_line.append(report_entry.id) %}
+{%         do csv_line.append(report_entry.device) %}
+{%         do csv_line.append(report_entry.test_category) %}
+{%         do csv_line.append(report_entry.test_description) %}
+{%         do csv_line.append(report_entry.test) %}
+{%         do csv_line.append(report_entry.result) %}
```

Comment:
you are missing arista.avd.default('') on all values. In case one of the values in any test is null the yaml_templates_to_facts will remove the key. So we will fail in this section.
Instead of faliing I think we should just output an empty field.
```suggestion
{%         do csv_line.append(report_entry.id | arista.avd.default("")) %}
{%         do csv_line.append(report_entry.device | arista.avd.default("")) %}
{%         do csv_line.append(report_entry.test_category | arista.avd.default("")) %}
{%         do csv_line.append(report_entry.test_description | arista.avd.default("")) %}
{%         do csv_line.append(report_entry.test | arista.avd.default("")) %}
{%         do csv_line.append(report_entry.result | arista.avd.default("")) %}
```